### PR TITLE
Add a 'homework' activity for people to do before the workshop 

### PR DESCRIPTION
**Pull request by aatwi**
1. Add a new Markdown "Homework.md" file under the docs folder
2. The md file includes the preparation required  to set up the dev environment

### CODE REVIEW

**Comment by aatwi** [comment]
I create a new MD file for now. Do you think it is better to have those steps included in the existing MD files?

@philou, I added a section for 'Pick your Constraint'. I thought it might be a good way to have all preparation in one MD 
file. WDYT?

**Review by AntoineMx — APPROVED** [review]

**Review by mengdaming — APPROVED** [review]

File: docs/Homework.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+   and processes to configure and run the project:
+    1. [C++ Build Manual](../cpp/BUILD_MANUAL.md)
+    1. [Java Build Manual](../java/BUILD_MANUAL.md)
+1. Your environment will be ready when you can: 
+    1. Open, compile, and run tests from your IDE
+    1. Run the application from the command-line (i.e., "Run Command" section of the BUILD_MANUAL)
+1. If you face any issue, you can refer to the [Troubleshooting](./Troubleshooting.md) page. 
+
+## Pick your Constraint 
+
```
Comment (philou) [inline docs/Homework.md:30]: Suggestion:

For the second iteration of the code retreat, you will have to pick a special constraint to try to fix the problem again. We propose 3 constraints:

* The Test Data Builders pattern: this technique let you factorize test data setup and make your tests more readable
* The Mikado Method: this technique lets you split large code changes in small independent steps that you can code, commit, integrate, and deploy independently, without blocking the feature delivery
* Test Data Builders + Mikado Method: mixing both techniques lets you write your tests incrementally, without blocking feature delivery and by increasing your overall code's testability at every step

Here are heuristics to pick the right constraint:

* Pick a technique you don't yet know
* If you know neither of the techniques, pick whichever you want, or just follow your group
* Don't try the combination of the 2 techniques if you do not already know the 2 techniques

**Review by philou — CHANGES_REQUESTED** [review]

File: docs/Homework.md (hunk 1 above)
Comment (aatwi) [reply docs/Homework.md:30]: Done. 
Do you think we need to reference the homework file from other files? or we just leverage on sending it to attendees ahead of time?

Comment (philou) [reply docs/Homework.md:30]: We could add a mention of it from the Facilitation guide.
But let's not wait before we send the homeworks to the registered murex attendees

Comment (aatwi) [reply docs/Homework.md:30]: Ok, I will merge it then :)

**Pull request merged**